edm4hep.yaml

```
@@ -227,3 +265,118 @@ datatypes :
+    Description: "LCIO vertex"
+    Author : "F.Gaede, DESY"
+    Members:
+      - int                 primary       //boolean flag, if vertex is the primary vertex of the event
+      - float               chi2          //chi-squared of the vertex fit
+      - float               probability   //probability of the vertex fit
+      - edm4hep::Vector3f   position      // [mm] position of the vertex.
+      - std::array<float,6> covMatrix     //covariance matrix of the position (stored as lower triangle matrix, i.e. cov(xx),cov(y,x),cov(z,x),cov(y,y),... )
+      - int                 algorithmType //type code for the algorithm that has been used to create the vertex - check/set the collection parameters AlgorithmName and AlgorithmType. 
+    VectorMembers:
+      - float               parameters    //additional parameters related to this vertex - check/set the collection parameter "VertexParameterNames" for the parameters meaning. 
+    OneToOneRelations:
+      - edm4hep::ReconstructedParticle   associatedParticle //reconstructed particle associated to this vertex.
+
+
+  #------- ReconstructedParticle
+  edm4hep::ReconstructedParticle:
+    Description: "Reconstructed Particle"
+    Author : "F.Gaede, DESY"
+    Members:
+      - int                    type           //type of reconstructed particle. Check/set collection parameters ReconstructedParticleTypeNames and ReconstructedParticleTypeValues.
+      - float                  energy         // [GeV] energy of the reconstructed particle.
+      - edm4hep::Vector3f      momentum       // [GeV] particle momentum
+      - edm4hep::Vector3f      referencePoint // [mm] reference, i.e. where the particle has been measured
+      - float                  charge         //charge of the reconstructed particle.
+      - float                  mass           // [GeV] mass of the reconstructed particle, set independently from four vector
+      - float                  goodnessOfPID  //overall goodness of the PID on a scale of [0;1]
+      - std::array<float,10>   covMatrix      //cvariance matrix of the reconstructed particle 4vector (10 parameters). Stored as lower triangle matrix of the four momentum (px,py,pz,E), i.e. cov(px,px), cov(py,##
+    OneToOneRelations:
+      - edm4hep::Vertex          startVertex    //start vertex associated to this particle
+      - edm4hep::ParticleID      particleIDUsed //particle Id used for the kinematics of this particle
+    OneToManyRelations:
+      - edm4hep::Cluster               clusters     //clusters that have been used for this particle. 
+      - edm4hep::Track                 tracks       //tracks that have been used for this particle. 
+      - edm4hep::ReconstructedParticle particles    //reconstructed particles that have been combined to this particle.
+      - edm4hep::ParticleID            particleIDs  //particle Ids (not sorted by their likelihood) 
+    ExtraCode:
+      declaration: "
+      bool isCompound() { return particles_size() > 0 ;}\n
```

```suggestion
      bool isCompound() { return particles.size() > 0 ;}\n
```
? Or is this some special podio construct?

---

Indeed, this is ok I think. Podio will generate a `${name}_size()` for these types of relations. I believe it comes from [here](https://github.com/AIDASoft/podio/blob/ad31c36f077487a1da130d3297ddcbb74f19e62a/python/templates/ConstRefVector.h.template). We should maybe also put the [generated code](https://gist.github.com/vvolkl/fc5d912ce6ba62840e512b4dfc89a4eb) for these classes on a doxygen page somewhere.